### ISSUE TITLE
New autocomplete component using downshift

### ISSUE THREAD

**Issue by steverydz** [canonical-web-and-design/react-components#620]
Referenced in https://github.com/canonical-web-and-design/snapcraft.io/pull/3709/files/1dac9743a7cd553ae40d78baa44431856b032011#r727302510

It would be good to have an autocomplete component, perhaps using https://github.com/downshift-js/downshift

This is used on the snapcraft brand store

**Comment by huwshimi** [comment]
Hi @steverydz do you think the datalist element would work for you, or are there more features you need? We recently started using datalist in MAAS.

https://vanillaframework.io/docs/examples/base/forms/datalist

**Comment by bartaz** [comment]
> The reason we would provide a wrapper around it is so that we can standardize the feature set, have consistent styling, and have a single place where people can come to pick 'blessed' components from for our projects. So I think even though it will end up being a very thin wrapper around the third party library it's still valuable for the team.

I agree, especially that it seems it has some specific styling designed, so we should turn it into Vanilla component, to ensure the styling is consistent across different uses of it.

Overall, as it sounds like a component with specific new styling it should be proposed to Vanilla, get reviewed by working group and implemented into Vanilla framework and React component here. 

Does downshift impose any constrains on the shape of the component (in terms of HTML structure of the search results for example)?

**Comment by clagom** [comment]
@AnnSoMuller a couple of examples:
- ubuntu.com > not implemented: https://zpl.io/VxNjdOk
- Juju Dashboard > part of the search and filter compoent:
![Screenshot 2022-03-31 at 09.22.31.png](https://images.zenhubusercontent.com/5b0d889d89d5dc41604fa14c/a40f97ff-81e5-4f5c-b5b9-7168ec116121)
Then once you click:
![Screenshot 2022-03-31 at 09.22.48.png](https://images.zenhubusercontent.com/5b0d889d89d5dc41604fa14c/bfb0c973-6628-45b1-98dd-9a395749f565)

**Comment by AnnSoMuller** [comment]
I have added the "autocomplete list" on the google document "Search & Filter pattern - documentation" [here](https://docs.google.com/document/d/1xvSeR0nmo71Po82OFBDA8x7NoS6OJN2Otxg9fIgVPCI/edit#heading=h.7pv7p4gt4v8d) for review

**Comment by davegoddard42** [comment]
@AnnSoMuller - this is great work! I've left a few very minor points on the doc. 

I noticed this is in the Search and Filter component specs, but the S&F already has a drop down functionality. I'm guessing this doesn't replace that? Should the functionality proposed actually fit in with the Search component, not the S&F? (It's so confusing with the naming...) Admittedly there isn't a Search specs doc yet - that's what I'm planning on doing this week - we can add this in too.

**Comment by AnnSoMuller** [comment]
@davegoddard42 Thanks for the feedback!
Indeed, It is to add to and nor replace the dropdown functionality :)

Good point too that I was first looking for the search spec and found out there is only the S&F one with the split part that explains you can split it :D
I've tried to stay simple and put the autocomplete where I thought I makes more sense.
Feel free to copy paste (or adapt!) that part into a search spec if you write it! I'll be happy to help or review.
Thanks!

**Comment by davegoddard42** [comment]
Awesome @AnnSoMuller. Yes, I will definitely add this into the Search specs doc. And I'll run it past you once written for review. Thanks

**Comment by cristinadresch** [comment]
@AnnSoMuller are the specs now finalised? could you paste the link here pls

**Comment by AnnSoMuller** [comment]
@cristinadresch I have posted the link above :)
But here it is => [Specs](https://docs.google.com/document/d/1xvSeR0nmo71Po82OFBDA8x7NoS6OJN2Otxg9fIgVPCI/edit#heading=h.7pv7p4gt4v8d)
I wanted to present (in DA and Vanilla WG) before closing it. If we agree, I'll post the adding bit in discourse 
Thanks!

**Comment by marinacastejon** [comment]
Reviewed on my end too. Thanks @AnnSoMuller !
